File: app/src/main/java/com/nichols/dsa/binarytrees/RightView.java
package com.nichols.dsa.binarytrees;
import com.nichols.dsa.binarytrees.DiameterOfTree.TreeNode;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;
import java.util.Queue;

public class RightView {
    public ArrayList<Integer> solveRightView(TreeNode A) {
        ArrayList<Integer> result = new ArrayList();
        rightView(A,result,0);
        return result;
    }

    public void rightView(TreeNode curr, List<Integer> result, int currDepth){
        if(curr == null)
            return;
        if(currDepth == result.size())
            result.add(curr.val);
        rightView(curr.right,result,currDepth+1);
        rightView(curr.left,result,currDepth+1);
    }
}

File: app/src/main/java/com/nichols/dsa/tree/RightView.kt
package com.nichols.dsa.tree

import com.nichols.dsa.binarytrees.DiameterOfTree.TreeNode

fun solvasdasde(A: TreeNode): IntArray {
    val result = arrayListOf<Int>()
    leftView(A, result, 0)
    return result.toIntArray()
}

fun leftView(curr: TreeNode?, result: ArrayList<Int>, currDepth: Int) {
    if (curr == null) {
        return
    }
    if (currDepth == result.size) {
        result.add(curr.`val`)
    }
    leftView(curr.left, result, currDepth + 1)
    leftView(curr.right, result, currDepth + 1)
}